src/ui/GenioWindow.cpp

```diff
@@ -1705,41 +1711,43 @@ GenioWindow::_FileOpen(BMessage* msg)
 
 		// new file to load..
 		selectTabInfo.AddBool("caret_position", true);
-		int32 index = fTabManager->CountTabs();
+		int32 index = fTabManager->SelectedTabIndex() + 1;
 		Editor* editor = _AddEditorTab(&ref, index, &selectTabInfo);
 
+		LogTrace("New index: %d, selected index: %d", index, fTabManager->SelectedTabIndex());
+
 		// TODO: using assert() is not nice, try to handle
 		// this gracefully if possible
 		assert(index >= 0);
```

Comment:
While you are there, Should these be turned into regular if statements?

Reply:
done